percy/providers/appAutomateProvider.js

```diff
@@ -72,35 +73,16 @@ class AppAutomateProvider extends GenericProvider {
     });
   }
 
-  async getDeviceName() {
-    return await TimeIt.run('getAADeviceName', async () => {
-      return (await this.getSessionDetails()).device;
-    });
-  }
-
-  async getOsVersion() {
-    return await TimeIt.run('getAAOsVersion', async () => {
-      return (await this.getSessionDetails()).os_version?.split('.')[0];
-    });
-  }
-
-  async getSessionDetails() {
-    return await Cache.withCache(Cache.bstackSessionDetails, this.driver.sessionId,
-      async () => {
-        return await TimeIt.run('getSessionDetails', async () => {
-          return await this.browserstackExecutor('getSessionDetails');
-        });
-      });
-  }
-
   async browserstackExecutor(action, args) {
     let options = args ? { action, arguments: args } : { action };
     return JSON.parse(await this.driver.execute(`browserstack_executor: ${JSON.stringify(options)}`));
   }
 
-  async getDebugUrl() {
-    return await TimeIt.run('getDebugUrl', async () => {
-      return (await this.getSessionDetails()).browser_url;
+  async setDebugUrl(result) {
+    return await TimeIt.run('setDebugUrl', async () => {
```

Comment:
This is not required to be done in `TimeIt.run`. We only use it for either high compute / high io operations. This is a constant time operation.